lib/brains/client.py

```diff
@@ -8,11 +9,31 @@
 log = getLogger('s.brains')
 
 
+class AuthEnvironment(braintree.environment.Environment):
+
+    def __init__(self, real):
+        self._url = urlparse(settings.BRAINTREE_PROXY)
+        self._real = real
+
+        super(self.__class__, self).__init__(
+            self._url.hostname, self._url.port, '',
+            self._url.scheme == 'https', None)
+
+
+AuthSandbox = AuthEnvironment(braintree.environment.Environment.Sandbox)
+AuthProduction = AuthEnvironment(braintree.environment.Environment.Production)
+
+
 class Http(braintree.util.http.Http):
 
-    def http_do(self, *args, **kw):
+    def http_do(self, verb, path, headers, body):
+        # Tell solitude-auth where we really want this request to go to.
+        headers['x-solitude-url'] = self.environment._real.base_url + path
+        # Set the URL of the request to point to the auth server.
+        path = self.env
+
         with statsd.timer('solitude.braintree.api'):
```

Comment:
Yeah good idea. I think everything gets prefixed so its in a different namespace, but will help confusion later on.